pegasus_wrapper/artifact.py

```diff
@@ -27,10 +27,16 @@ class DependencyNode:
     """
 
     job: Optional[Job] = attrib(validator=optional(instance_of(Job)), kw_only=True)
+    output_files: ImmutableSet[File] = attrib(
```

Comment:
Can you add a comment describing what this is for? (I know, but future readers might not)